packages/core/src/carousel/private/utils/withHorizontalScroll.ts

```diff
@@ -232,6 +233,32 @@ export const withHorizontalScroll = (options: Options): any => {
     }
   }, []);
 
+  const handleMouseDown = () => {
+    setIsDragging((prevState) => !prevState);
+  };
+
+  const handleMouseUp = () => {
+    setIsDragging((prevState) => !prevState);
+  };
+
+  const debounce = (fallback: (...args: any) => void, delay: number) => {
+    let timer: number;
+    return (...args: any) => {
+      clearTimeout(timer);
+      timer = setTimeout(() => {
+        fallback(...args);
+      }, delay);
+    };
+  };
+
+  const debounceMouseDown = () => {
```

Comment:
debounceMouseDown and debounceMouseUp might be good to have wrapped with useMemo, or moved outside. Potential performance issue.

Reply:
Agreed. Updated. Thanks for pointing up.